spec/services/feeds/assemble_article_markdown_spec.rb

```diff
@@ -0,0 +1,51 @@
+require "rails_helper"
+
+RSpec.describe Feeds::AssembleArticleMarkdown, type: :service do
+  let(:user) { create(:user, feed_mark_canonical: true) }
+  let(:feed_source_url) { "https://feed.source/url" }
+  let(:feed) { Struct.new(:url).new("https://feed.source/") }
+  let(:title) { "A title" }
+  let(:content) { "Some content that came in with the item, should be the body" }
+
+  let(:item) do
+    Struct
```

Comment:
pretty sure this should be a double instead of a struct.

Reply:
Let's make it a verifying double then 😃 This requires some minor code changes but I honestly think they are for the better as the current code mixes method access and `:[]` access for `Feedjira::Parser::RSSEntry`:

```diff
diff --git a/app/services/feeds/assemble_article_markdown.rb b/app/services/feeds/assemble_article_markdown.rb
index 75325751b..5fc6abde1 100644
--- a/app/services/feeds/assemble_article_markdown.rb
+++ b/app/services/feeds/assemble_article_markdown.rb
@@ -6,8 +6,8 @@ def self.call(item, user, feed, feed_source_url)

     def initialize(item, user, feed, feed_source_url)
       @item = item
-      @title = item[:title].strip
-      @categories = item[:categories] || []
+      @title = item.title.strip
+      @categories = item.categories || []
       @user = user
       @feed = feed
       @feed_source_url = feed_source_url
@@ -58,7 +58,7 @@ def assemble_body_markdown
     end

     def get_content
-      @item[:content] || @item[:summary] || @item[:description]
+      @item.content || @item.summary || @item.description
     end

     def thorough_parsing(content, feed_url)
diff --git a/spec/services/feeds/assemble_article_markdown_spec.rb b/spec/services/feeds/assemble_article_markdown_spec.rb
index 4365b4e81..af44fcacf 100644
--- a/spec/services/feeds/assemble_article_markdown_spec.rb
+++ b/spec/services/feeds/assemble_article_markdown_spec.rb
@@ -3,19 +3,18 @@
 RSpec.describe Feeds::AssembleArticleMarkdown, type: :service do
   let(:user) { create(:user, feed_mark_canonical: true) }
   let(:feed_source_url) { "https://feed.source/url" }
-  let(:feed) { Struct.new(:url).new("https://feed.source/") }
+  let(:feed) { instance_double("Feedjira::Parser::RSS", url: "https://feed.source/") }
   let(:title) { "A title" }
   let(:content) { "Some content that came in with the item, should be the body" }

   let(:item) do
-    Struct
-      .new(:title, :categories, :published, :content)
-      .new(
-        title,
-        %w[tag1 tag2 tag3 tag4 tag5],
-        "2020-12-20",
-        content,
-      )
+    instance_double(
+      "Feedjira::Parser::RSSEntry",
+      title: title,
+      categories: %w[tag1 tag2 tag3 tag4 tag5],
+      published: "2020-12-20",
+      content: content,
+    )
   end

   let(:feeds_assemble_article_markdown) { described_class.new(item, user, feed, feed_source_url) }
```

The specs are green for me with all these changes applied.

Reply:
Thanks for digging this out!